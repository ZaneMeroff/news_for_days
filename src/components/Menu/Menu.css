
nav {
  position: fixed;
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 15%;
  height: 100vh;
  padding: 1%;
  background-color: rgb(59, 98, 131, 1);
}

.logo-title {
  font-family: Bangers;
  font-size: 200%;
  margin-bottom: 10%;
}

.nav-button {
  border-radius: 15px;
  color: #000000;
  width: 90%;
  margin: 3%;
  height: 30px;
}

.nav-button.selected {
  background-color: #82c6ff;
  border: 2px solid #e85a5a;
}

.nav-button:hover {
  cursor: pointer;
  transform: scale(1.08);
  transition: transform .5s;
}
