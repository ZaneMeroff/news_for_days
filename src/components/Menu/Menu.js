import React from 'react';
import './Menu.css';

const Menu = ({setNewsType, selectedNewsType}) => {

  const selectNewsType = (newsType) => {
    setNewsType(newsType);
  }

  const newsTypes = ['local', 'health', 'entertainment', 'science', 'technology'];

  return (
    <nav>
      <h1 className='logo-title'>Read all about it!</h1>
      {newsTypes.map(news => {
       return (
         <button key={Math.random()} className={news === selectedNewsType ? 'nav-button selected' : 'nav-button'} onClick={() => selectNewsType(news)}>{news.toUpperCase()}</button>
       )
      })}
    </nav>
  );
}

export default Menu;
