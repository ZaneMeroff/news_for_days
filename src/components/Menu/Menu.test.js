import React from 'react';
import Menu from './Menu'
import { shallow } from 'enzyme';

describe('Menu', () => {

  it('should be able to render menu to page', () => {
    const wrapper = shallow(<Menu
      setNewsType={jest.fn()}
      selectedNewsType={'local'}
      />)
    expect(wrapper).toMatchSnapshot();
  });

  it('expect selectNewsType to be called on button click', () => {
    const mockSetNewsType = jest.fn();
    const wrapper = shallow(<Menu
      setNewsType={mockSetNewsType}
      selectedNewsType={'local'}
      />)
    wrapper.find('button').at(0).simulate('click')
    expect(mockSetNewsType).toHaveBeenCalled();
  });

});
