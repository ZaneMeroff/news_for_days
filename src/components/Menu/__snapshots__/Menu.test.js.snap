// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should be able to render menu to page 1`] = `
<nav>
  <h1
    className="logo-title"
  >
    Read all about it!
  </h1>
  <button
    className="nav-button selected"
    key="local"
    onClick={[Function]}
  >
    LOCAL
  </button>
  <button
    className="nav-button"
    key="health"
    onClick={[Function]}
  >
    HEALTH
  </button>
  <button
    className="nav-button"
    key="entertainment"
    onClick={[Function]}
  >
    ENTERTAINMENT
  </button>
  <button
    className="nav-button"
    key="science"
    onClick={[Function]}
  >
    SCIENCE
  </button>
  <button
    className="nav-button"
    key="technology"
    onClick={[Function]}
  >
    TECHNOLOGY
  </button>
</nav>
`;
